Jenkinsfile

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env groovy
+
+// Application-specific Values
+def mavenArgs="-Dcom.redhat.xpaas.repo.redhatga"     // Global maven arguments
+def mavenPackageArgs="package spring-boot:repackage" // Maven package arguments
+def mavenOutputJar="parksmap-web.jar"                // Output jar from Maven
+def appName="parksmap"                               // Application name (for route, buildconfig, deploymentconfig)
+def previewAppName="parksmap"                        // Application used for viewing/testing
+
+
+// Pipeline variables
+def isPR=false                   // true if the branch being tested belongs to a PR
+def baseProject=""               // name of base project - used when testing a PR
+def project=""                   // project where build and deploy will occur
+def projectCreated=false         // true if a project was created by this build and needs to be cleaned up
+def repoUrl=""                   // the URL of this project's repository
+
+// uniqueName returns a name with a 16-character random character suffix
+def uniqueName = { String prefix ->
+  sh "cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 16 | head -n 1 > suffix"
+  suffix = readFile("suffix").trim()
+  return prefix + suffix
+}
+
+// setBuildStatus sets a status item on a GitHub commit
+def setBuildStatus = { String url, String context, String message, String state, String backref ->
+  step([
+    $class: "GitHubCommitStatusSetter",
+    reposSource: [$class: "ManuallyEnteredRepositorySource", url: url ],
+    contextSource: [$class: "ManuallyEnteredCommitContextSource", context: context ],
+    errorHandlers: [[$class: "ChangingBuildStatusErrorHandler", result: "UNSTABLE"]],
+    statusBackrefSource: [ $class: "ManuallyEnteredBackrefSource", backref: backref ],
+    statusResultSource: [ $class: "ConditionalStatusResultSource", results: [
+        [$class: "AnyBuildResult", message: message, state: state]] ]
+  ]);
+}
```

Comment:
Agreed. It'd also be nice if we have the capability to:
- add comments
- merge